src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,172 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private static class Node<E> {
```

Comment:
Lets declare MyLinkedList class fields first, and then define class Node<T>
```suggestion
    private static class Node<T> {
```